src/mapgen.cpp

```diff
@@ -13343,6 +13343,15 @@ void map::add_extra(map_extra type)
             }
             add_trap(x, y, tr_landmine_buried);
         }
+        int x1 = 0, y1 = 0, x2 = (SEEX * 2 - 1), y2 = (SEEY * 2 - 1);
```

Comment:
I'd suggest putting these on separate lines just to make things a bit clearer, and in the section above, I'd make it so it couldn't spawn mines along the sides of the tile, or at least in the corners.